src/main/scala/za/co/absa/hyperdrive/trigger/persistance/WorkflowRepository.scala

```diff
@@ -144,18 +144,28 @@ class WorkflowRepositoryImpl extends WorkflowRepository {
       }
   }
 
-  override def updateWorkflowActiveState(id: Long, isActive: Boolean)(implicit ec: ExecutionContext): Future[Unit] = db.run(
-    for {
-      w <- workflowTable.filter(_.id === id)
-        .map(workflow => (workflow.isActive, workflow.updated)).update((isActive, Option(LocalDateTime.now())))
+  override def switchWorkflowActiveState(id: Long)(implicit ec: ExecutionContext): Future[Unit] = {
+    val targetWorkflow = workflowTable.filter(_.id === id).map(workflow => (workflow.isActive, workflow.updated))
```

Comment:
Took me some time to figure this for-comprehension out. Maybe renaming helps
`targetWorkflow` -> `workflowQuery`, `workflowToUpdateOption` -> `workflowOpt`. Just a suggestion. But I would only use either the word "target" or "toUpdate" because in this context they have the same meaning.

Reply:
Done